src/context-selection/period-selector-bar-item/use-periods.js

```diff
@@ -0,0 +1,79 @@
+import moment from 'moment'
+import { useMemo } from 'react'
+import {
+    YEARLY,
+    FINANCIAL_APRIL,
+    FINANCIAL_JULY,
+    FINANCIAL_OCT,
+    FINANCIAL_NOV,
+    addFullPeriodTimeToDate,
+    getCurrentDate,
+    getFixedPeriodsForTypeAndDateRange,
+    getYearlyPeriodIdForTypeAndYear,
+    parsePeriodId,
+} from '../../shared/index.js'
+
+const yearlyPeriodTypes = [
+    YEARLY,
+    FINANCIAL_APRIL,
+    FINANCIAL_JULY,
+    FINANCIAL_OCT,
+    FINANCIAL_NOV,
+]
+
+export default function usePeriods({
+    periodType,
+    futurePeriods,
+    openFuturePeriods,
+    year,
+}) {
+    return useMemo(() => {
+        if (!periodType) {
+            return []
+        }
+
+        let periods
+        const currentDate = getCurrentDate()
+        const currentYear = currentDate.getFullYear()
+        const relevantFuturePeriods = futurePeriods.filter((futurePeriod) => {
+            return (
+                new Date(futurePeriod.startDate).getFullYear() === currentYear
+            )
+        })
+
+        if (yearlyPeriodTypes.includes(periodType)) {
+            const yearsCount =
+                getCurrentDate().getFullYear() - 1970 + openFuturePeriods
+            periods = Array.from(Array(yearsCount)).map((_, index) => {
+                const year = index + 1970
+                const periodId = getYearlyPeriodIdForTypeAndYear(
+                    periodType,
+                    year
+                )
+                return parsePeriodId(periodId)
+            })
+        } else if (year > currentYear) {
+            return futurePeriods.filter(
```

Comment:
this should be `periods = futurePeriods.filter(//etc)` as currently the future periods are not being reversed (whereas the current/past periods are reversed)